### DISCUSSION TITLE
Automatic Grouping: Every search creates a group based on keywords (if no group exists).

### DISCUSSION THREAD

**Discussion by FreePDF64** [okaryo/TabTabTab#408] [Ideas]
Automatic Grouping: Every search creates a group based on keywords (if no group exists).

**Comment by okaryo** [comment]
@FreePDF64 
Hi,

Is my understanding correct that this feature would, for example, group tabs displayed in the search results when searching for `GitHub` (e.g., by adding a "Group Tabs" button on the search results page)? If so, this sounds like a very useful feature. Thank you for the suggestion!

**Reply by FreePDF64** [reply]
Yes, your understanding is correct! :)

**Comment by FreePDF64** [comment]
I think this feature is a killer feature that hardly any other tab manager offers.

👍

**Comment by okaryo** [comment]
@FreePDF64 
Hi,

I’ve implemented this feature in v2.36.0, and it’s currently under review in the store. Once it’s released, I’d love for you to try it out. If you encounter any issues or have suggestions for improvement, please feel free to share them.

**Comment by FreePDF64** [comment]
I'm really excited to test this new feature. Thank you in advance!

**Comment by FreePDF64** [comment]
Hi,

I don't know how it works. Nothing happend by me after click of the blue button "Group Tabs: xxx".
I expected that a new tab group would be created in the active browser with the tabs I was looking for.
Please explain :)

Note: Has been tested with Brave Browser and Edge Browser

Kind regards,
Michael

**Reply by okaryo** [reply]
Hi,

I’m sorry about the issue you encountered. It seems there was a bug causing the behavior you described. I’ve fixed it and will release an updated version shortly. Thank you for bringing this to my attention, and I appreciate your patience!

**Comment by FreePDF64** [comment]
Looks almost perfect! 
- An additional feature here would be if you could also adjust the name of the group before the group is created based on the search query.
- After clicking “Create Group” you should return to the original home page of TabTabTab: the view of all tabs.

**Reply by okaryo** [reply]
Thank you for checking it so quickly!

>An additional feature here would be if you could also adjust the name of the group before the group is created based on the search query.

That’s a great idea. However, I can’t immediately think of the best UI for this. I’ll work on it once I come up with a good solution.

>After clicking “Create Group” you should return to the original home page of TabTabTab: the view of all tabs.

You’re absolutely right. I’ll address this by next week.

Thank you for all the valuable feedback!

**Reply by FreePDF64** [reply]
You may get some ideas from this AddOn: https://chromewebstore.google.com/detail/notab-vertical-tab-and-gr/moldjoddllhdbbpimnegpiephdcanmia

**Reply by okaryo** [reply]
Thanks! I'll check it.

**Comment by multi-suggester** [comment]
maybe an sub option for only grouping the ungrouped tabs and a button for reversing the process maybe
 (ive lost my order of 540 tabs ;( ) (i know its saveable by re opening the tab groups but ill lost tab histories if i do that etc etc)
 
 (i think if you were to do this i think easiest way would be trying to return to the last screenshot by reversing an action , or just trying to return to last screenshot by current ingredients sort of thing)
